test/mapper_text_test.rb

```diff
@@ -0,0 +1,136 @@
+      result[1].source.must_equal '_word_'
+      result[1].text.must_equal 'word'
+      result[1].class.name.must_equal 'Asciidoctor::InlineParser::Emphasis'
+    end
+    it 'should map an unconstrained emphasis letter' do
+      input = 'This is a sentence with an emphasis le__t__ter'
+      ast = ::Asciidoctor::InlineParser.raw_parse input
+      result = ::Asciidoctor::InlineParser::Mapper.map ast
+      result.size.must_equal 3
+      result.first.source.must_equal 'This is a sentence with an emphasis le'
+      result[1].source.must_equal '__t__'
+      result[1].text.must_equal 't'
+      result[1].class.name.must_equal 'Asciidoctor::InlineParser::Emphasis'
+      result[2].source.must_equal 'ter'
+    end
+  end
+  describe 'simple sentence with a constrained and unconstrained monospaced' do
+    it 'should map a constrained monospaced word' do
+      input = 'This is a sentence with a monospaced `word`'
+      ast = ::Asciidoctor::InlineParser.raw_parse input
+      result = ::Asciidoctor::InlineParser::Mapper.map ast
+      result.size.must_equal 2
+      result.first.source.must_equal 'This is a sentence with a monospaced '
+      result[1].source.must_equal '`word`'
+      result[1].text.must_equal 'word'
+      result[1].class.name.must_equal 'Asciidoctor::InlineParser::Code'
+    end
+    it 'should map an unconstrained monospaced letter' do
+      input = 'This is a sentence with a monospaced le``t``ter'
+      ast = ::Asciidoctor::InlineParser.raw_parse input
+      result = ::Asciidoctor::InlineParser::Mapper.map ast
+      result.size.must_equal 3
+      result.first.source.must_equal 'This is a sentence with a monospaced le'
+      result[1].source.must_equal '``t``'
+      result[1].text.must_equal 't'
+      result[1].class.name.must_equal 'Asciidoctor::InlineParser::Code'
+      result[2].source.must_equal 'ter'
+    end
+  end
+  it 'should map a simple sentence with a superscript single letter' do
+    input = 'This is a famous formula: E = mc^2^'
+    ast = ::Asciidoctor::InlineParser.raw_parse input
+    result = ::Asciidoctor::InlineParser::Mapper.map ast
+    result.size.must_equal 2
+    result.first.source.must_equal 'This is a famous formula: E = mc'
+    result[1].source.must_equal '^2^'
+    result[1].text.must_equal '2'
+    result[1].class.name.must_equal 'Asciidoctor::InlineParser::Superscript'
+  end
+  it 'should map a simple sentence with a subscript single letter' do
+    input = 'This is a subscript ~a~'
+    ast = ::Asciidoctor::InlineParser.raw_parse input
+    result = ::Asciidoctor::InlineParser::Mapper.map ast
+    result.size.must_equal 2
+    result.first.source.must_equal 'This is a subscript '
+    result[1].source.must_equal '~a~'
+    result[1].text.must_equal 'a'
+    result[1].class.name.must_equal 'Asciidoctor::InlineParser::Subscript'
+  end
+  describe 'nested' do
+    it 'should map a nested formatted string (emphasis text inside a bold text)' do
```

Comment:
@mojavelinux Here's the interesting part, the input is `The next *few words are _really_ important!*` and the AST (represented as JSON) is:

```js
[
  {
    type: 'text',
    source: 'The next'
  },
  {
    type: 'strong',
    source: '*few words are _really_ important!*',
    text: 'few words are _really_ important!',
    children: [
      {
        type: 'text',
        source: 'few words are '
      },
      {
        type: 'emphasis',
        source: '_really_',
        text: 'really'
      },
      {
        type: 'text',
        source: ' important!'
      }
    ]
  }
]
```

Does this representation look OK ?